### PR TITLE
[DOC][SQL] update out-of-date code snippets using SQLContext in all documents.

### PR DESCRIPTION
**Pull request by WeichenXu123**
## What changes were proposed in this pull request?

I search the whole documents directory using SQLContext, and update the following places:
- docs/configuration.md, sparkR code snippets.
- docs/streaming-programming-guide.md, several example code.
## How was this patch tested?

N/A

### CODE REVIEW

**Comment by WeichenXu123** [comment]
cc @srowen

**Comment by WeichenXu123** [comment]
cc @liancheng

**Comment by liancheng** [comment]
@WeichenXu123 Is this ready for review? If yes, please remove the WIP tag in the PR description.

**Comment by WeichenXu123** [comment]
@liancheng Yes. Thanks!

File: docs/configuration.md [hunk 1]
```diff
@@ -1564,8 +1564,8 @@ spark.sql("SET -v").show(n=200, truncate=False)
 <div data-lang="r"  markdown="1">
 
 {% highlight r %}
-# sqlContext is an existing sqlContext.
-properties <- sql(sqlContext, "SET -v")
+# spark is an existing SparkSession.
+properties <- sql(spark, "SET -v")
```
Comment (liancheng) [inline docs/configuration.md]: I believe in the most recent SparkR API, you don't need to pass in the `spark` parameter to run a SQL statement now.

**Comment by liancheng** [comment]
@tdas Would you please to help review streaming example code changes? Thanks!

File: docs/configuration.md (hunk 1 above)
Comment (WeichenXu123) [inline docs/configuration.md]: @liancheng Fix it, thanks!

File: docs/streaming-programming-guide.md [hunk 2]
```diff
@@ -1546,9 +1546,9 @@ val words: DStream[String] = ...
 
 words.foreachRDD { rdd =>
 
-  // Get the singleton instance of SQLContext
-  val sqlContext = SQLContext.getOrCreate(rdd.sparkContext)
-  import sqlContext.implicits._
+  // Get the singleton instance of SparkSession
+  val spark = SparkSessionSingleton.getInstance(rdd.sparkContext.getConf)
```
Comment (rxin) [inline docs/streaming-programming-guide.md]: SparkSessionSingleton is only available in an example file -- i am not sure we can use it in the doc here.

Comment (WeichenXu123) [inline docs/streaming-programming-guide.md]: @rxin Now I modify the doc example code using raw `SparkSession` API. Thanks!

**Comment by rxin** [comment]
Thanks - merging in master/2.0.

**Pull request closed without merging**